nrf_rpc/include/nrf_rpc.h

```diff
@@ -73,7 +67,7 @@ struct nrf_rpc_decoder {
  * Created by @a NRF_RPC_GROUP_DEFINE.
  */
 struct nrf_rpc_group {
-	uint8_t group_id;
+	uint8_t *group_id;
```

Comment:
In current implementation yes, but we have to prepare that nRF RPC will be able to do communication between chip and PC. It is very uncommon to change linker script when compiling for Windows or Linux. I have draft of NRF_RPC_AUTO_ARR implementation based on `__attribute__((constructor))` instead of sections. In this implementation such pointer arithmetic will not work.

As I can see it we have two options:
1. Keep is as it is,
2. Add another macro (e.g. `NRF_RPC_AUTO_ARR_INDEX(array, item_ptr)`) that will return array index based on provided pointer to the item. Section based implementation will look like you provided. `__attribute__((constructor))` based implementation will be a little more complicated, but acceptable if primary target will be PC.

Reply:
@trond-snekvik, can you gave a feedback regarding options that I suggested above?

Reply:
Keep it as is 👍